### PR TITLE
Refactored src/components/AgendaCategory/AgendaCategoryContainer.test.tsx from jest to vitest

### LINKED ISSUE #2791: Refactor `src/components/AgendaCategory/AgendaCategoryContainer.test.tsx` from Jest to Vitest
**Is your feature request related to a problem? Please describe.**

This files in this directory and all its subdirectories are currently tested using Jest. As part of our migration to Vitest, we need to refactor all these files' test cases to align with Vitest's syntax and features.

**Acceptance Criteria**

For each `*.test.ts` and `*.test.tsx` file found under the parent:

1. Replace Jest-specific functions and mocks with Vitest equivalents
1. Rename the test file from having a `.test.*` suffix to  a `.spec.*` suffix
1. Ensure all tests in the file pass after migration using `npm run test:vitest`
1. Maintain the test coverage for the file as 100% after migration
1. Upload a video or photo for this specific file coverage is 100% in the PR description

**Additional context**

- Refer to the merged PR for the base Vitest configuration:
https://github.com/PalisadoesFoundation/talawa-admin/pull/2457/files#diff-9a5a35cf8d7f02b2bba90905714b762560579c4789e9eb4cebb4607980aa3d93
- Vitest documentation:
https://vitest.dev/

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by AceHunterr**
**What kind of change does this PR introduce?**
Refactoring

**Issue Number:**
Fixes #2791 

**Did you add tests for your changes?**
No

**Snapshots/Videos:**
Here is the uploaded Snapshots for the successful test runs:

img2
img1

**If relevant, did you update the documentation?**
No

**Summary**

I open a PR earlier regarding the same issue... but that was having some troubles regarding the file changes in the CheckIn template... so pull and created a new PR for the issue.

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

## Summary by CodeRabbit

- **Tests**
	- Transitioned from Jest to Vitest for testing functionalities in the AgendaCategory components.
	- Updated mocking functions to use `vi.fn()` instead of `jest.fn()`.
	- Adjusted import statements to incorporate Vitest's testing functions.

### CODE REVIEW

File: src/components/AgendaCategory/AgendaCategoryContainer.spec.tsx [hunk 1]
```diff
@@ -8,9 +8,9 @@ import {
   fireEvent,
 } from '@testing-library/react';
 import userEvent from '@testing-library/user-event';
-import 'jest-localstorage-mock';
+// import 'jest-localstorage-mock';
```
Comment (palisadoes) [inline src/components/AgendaCategory/AgendaCategoryContainer.spec.tsx:11]: Delete versus commenting

File: src/components/AgendaCategory/AgendaCategoryContainer.spec.tsx [hunk 2]
```diff
@@ -8,9 +8,9 @@ import {
...
 import { MockedProvider } from '@apollo/client/testing';
-import 'jest-location-mock';
+// import 'jest-location-mock';
```
Comment (palisadoes) [inline src/components/AgendaCategory/AgendaCategoryContainer.spec.tsx:13]: 31bb6b

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comments

**Comment by AceHunterr** [comment]
Rectified that

**Review by palisadoes — APPROVED** [review]

**Pull request merged**